### PR TITLE
[7.5.x] RHDM-163: Teams icons show number of repositories

### PR DESCRIPTION
**Pull request by karreiro**
See: https://issues.jboss.org/browse/RHDM-163

---

demo

### CODE REVIEW

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/screens/organizationalunit/OrganizationalUnitsScreen.java [hunk 1]
```diff
@@ -114,31 +104,61 @@ private void setupView() {
...
-                            String.valueOf(organizationalUnit.getRepositories().size()),
-                            view.getNumberOfRepositoriesLabel(organizationalUnit.getRepositories().size()),
+                            String.valueOf(numberOfProjects),
+                            view.getNumberOfProjectsLabel(numberOfProjects),
                             () -> open(organizationalUnit));
             view.addOrganizationalUnit(tileWidget);
         });
     }
 
+    int getNumberOfProjects(final OrganizationalUnit organizationalUnit) {
```
Comment (jomarko) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/screens/organizationalunit/OrganizationalUnitsScreen.java]: Could we this implement as below, or am I missing something:
```java
final Integer count = numberOfProjectsByOrganizationalUnit.get(organizationalUnit);
return count != null ? count.intValue() : 0;
```

**Comment by jomarko** [comment]
Tried manually, I can confirm the code works. Please see my review above before merging.

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-backend/src/main/java/org/kie/workbench/common/screens/impl/LibraryServiceImpl.java [hunk 2]
```diff
@@ -365,6 +367,60 @@ public Project importProject(final OrganizationalUnit organizationalUnit,
         return new ArrayList<>(ouService.getOrganizationalUnits());
     }
 
+    @Override
+    public Map<OrganizationalUnit, Integer> getNumberOfProjectsByOrganizationalUnit(final List<OrganizationalUnit> organizationalUnits) {
+
+        final Map<OrganizationalUnit, Integer> numberOfProjectsByOrganizationalUnit = new HashMap<>();
```
Comment (jhrcek) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-backend/src/main/java/org/kie/workbench/common/screens/impl/LibraryServiceImpl.java]: (Optional) You're using streams throughout this PR, so you could use them here too, for consistency's sake

```
return organizationalUnits.stream()
                .collect(toMap(Function.identity(), ou -> numberOfProjects(ou)));
```

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-backend/src/main/java/org/kie/workbench/common/screens/impl/LibraryServiceImpl.java [hunk 3]
```diff
@@ -365,6 +367,60 @@ public Project importProject(final OrganizationalUnit organizationalUnit,
...
+        return numberOfProjectsByOrganizationalUnit;
+    }
+
+    private int numberOfProjects(final OrganizationalUnit organizationalUnit) {
+        return getProjects(organizationalUnit).size();
+    }
+
+    Set<String> getProjects(final OrganizationalUnit organizationalUnit) {
+
+        final Set<String> projects = new HashSet<>();
```
Comment (jhrcek) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-backend/src/main/java/org/kie/workbench/common/screens/impl/LibraryServiceImpl.java]: (Optional) ... and here.

```
return getAuthorizedRepositories(organizationalUnit).stream()
                .flatMap(repo -> getProjects(repo).stream())
                .collect(toSet());|
```

**Comment by karreiro** [comment]
Thank you for the review, @jomarko. All changes were applied.

PS: Regarding the variables names, preferred to rename only the variables in the added code and keep the fields intact.

---

Thank you for the review too @jhrcek.

**Pull request merged**